### PR TITLE
Declare dependencies explicitly

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this is BC.

I consider this fixes this other PR https://github.com/sonata-project/dev-kit/pull/385 (because that was related to this bundle)

## Subject

This changes nothing on what gets installed when using this bundle. But declaring explicit dependencies makes you use the versions correctly even when doing prefer-lowest. That makes the build faster, and improves the consistency on this bundle, because we know what is needed for this bundle to work exactly.

Labelled as pedantic cause it will change nothing for normal users

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,12 +25,20 @@
     "require": {
         "php": "^5.6 || ^7.0",
         "egeloen/ckeditor-bundle": "^4.0 || ^5.0",
-        "knplabs/knp-markdown-bundle": "^1.4",
+        "knplabs/knp-markdown-bundle": "^1.7",
         "sonata-project/block-bundle": "^3.11",
-        "sonata-project/core-bundle": "^3.0",
+        "sonata-project/core-bundle": "^3.9",
+        "symfony/config": "^2.8 || ^3.2 || ^4.0",
```
Comment (kunicmarko20) [inline composer.json]: You can probably then increase Symfony to `^3.3` also.

File: composer.json [hunk 2]
```diff
@@ -25,12 +25,20 @@
     "require": {
         "php": "^5.6 || ^7.0",
         "egeloen/ckeditor-bundle": "^4.0 || ^5.0",
-        "knplabs/knp-markdown-bundle": "^1.4",
+        "knplabs/knp-markdown-bundle": "^1.7",
```
Comment (OskarStark) [inline composer.json]: why this change?

Comment (jordisala1991) [reply composer.json]: Because I was trying to reduce the matrix of possible options when doing prefer-lowest to have faster builds

Comment (OskarStark) [reply composer.json]: 👍

**Comment by kunicmarko20** [comment]
Thanks @jordisala1991 !
